packages/front-end/components/Experiment/MetricsSelector.tsx

```diff
@@ -1,24 +1,26 @@
 import { FC } from "react";
 import { FaQuestionCircle } from "react-icons/fa";
 import { useDefinitions } from "@/services/DefinitionsContext";
-import Tooltip from "../Tooltip/Tooltip";
-import MultiSelectField from "../Forms/MultiSelectField";
-import SelectField from "../Forms/SelectField";
+import MultiSelectField from "@/components/Forms/MultiSelectField";
+import SelectField from "@/components/Forms/SelectField";
+import Tooltip from "@/components/Tooltip/Tooltip";
 
 const MetricsSelector: FC<{
   datasource?: string;
+  project?: string;
   selected: string[];
   onChange: (metrics: string[]) => void;
   autoFocus?: boolean;
-}> = ({ datasource, selected, onChange, autoFocus }) => {
+}> = ({ datasource, project, selected, onChange, autoFocus }) => {
   const { metrics } = useDefinitions();
-
-  const validMetrics = metrics.filter(
-    (m) => !datasource || m.datasource === datasource
-  );
+  const filteredMetrics = metrics
+    .filter((m) => m.datasource === datasource)
+    .filter(
+      (m) => project && m?.projects?.length && m.projects.includes(project)
```

Comment:
This will break if project is undefined or the metric isn't tagged to any projects.